File: compiler/testdata/ruby/test_class_methods.rb
class Parent
  @@family_things = []    # Shared between class and subclasses
  @shared_things  = []    # Specific to this class

  def self.family_things
    @@family_things
  end
  def self.shared_things
    @shared_things
  end

  attr_accessor :my_things
  def initialize
    @my_things = []       # Just for me
  end
  def family_things
    self.class.family_things
  end
  def shared_things
    self.class.shared_things
  end
end

class Child < Parent
  @shared_things = []
end

File: compiler/testdata/go/test_class_methods.go
package main

type parentClass struct {
	shared_things []string
}

func (p *parentClass) Family_things() []string {
	// @@family_things
}
func (p *parentClass) Shared_things() []string {
	return p.shared_things
}

var ParentClass = &parentClass{}

type Parent struct {
	Class    *parentClass
	MyThings []string
}

func NewParent() *Parent {
	return &Parent{}
}

func (p *Parent) Family_things() []string {
	// self.class.family_things
}
func (p *Parent) Shared_things() []string {
	return p.Class.Shared_things()
}

type Child struct {
	MyThings []string
}

func NewChild() *Child {
	return &Child{}
}

func (p *Child) Family_things() []string {
	// self.class.family_things
}
func (p *Child) Shared_things() []string {
	// self.class.shared_things
}
